src/agent/agent_instance.hpp

```diff
@@ -66,11 +66,14 @@ class AgentInstance
     /**
      * This method initialize the agent.
      *
+     * @param[in] aThreadIfName    The Thread network interface name.
+     * @param[in] aBackboneIfName  The Backbone network interface name.
+     *
      * @retval  OTBR_ERROR_NONE     Agent initialized successfully.
      * @retval  OTBR_ERROR_ERRNO    Failed due to error indicated in errno.
      *
      */
-    otbrError Init(void);
+    otbrError Init(const std::string &aThreadIfName, const std::string &aBackboneIfName);
```

Comment:
why using std::string?

Reply:
We may also use `const char *`. But I don't see the disadvantage of using `std::string` compared with `const char *`. 
Thoughts?

Reply:
Changed to use `const InstanceParams &aInstanceParams` as suggested by @librasungirl . 
In `InstanceParams`, the Thread and Backbone names are defines as member char arrays.